src/hash-table.c

```diff
@@ -32,8 +32,9 @@ CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 #endif
 
 struct _HashTableEntry {
-	HashTableKey key;
-	HashTableValue value;
+//	HashTableKey key;
+//	HashTableValue value;
+	KeyValuePair* value_pair;
```

Comment:
Also: I don't think it's even necessary for this to be a pointer. 'value_pair's lifetime is intrinsically linked to HashTableEntry so there's no need for them to be allocated separately - it just wastes memory.